AprilSylph/XKit-Rewritten
# feat(PostBlock): Implement New XKit migration

### Description


#1279 but for PostBlock.

Along with https://github.com/new-xkit/XKit/pull/2182, this implements a button that migrates PostBlock data from New XKit to XKit Rewritten when pressed. It communicates via custom events from one extension's content script to the other. 

### Testing steps


- Load the corresponding New XKit PR (one can check out the branch before installing PostBlock or can force update PostBlock).
- Use New XKit to block at least one post.
- Enable PostBlock in XKit Rewritten.
- Press the button in the New XKit PostBlock settings pane and confirm the migration.
- Unblock the post in New XKit. Refresh the page. (Apparently we never added cleanup to XKit 7 PostBlock, whee).
- Confirm that the post is hidden, and that it appears again when the last entry is removed from XKit Rewritten PostBlock.
